lib/file_controller.dart

```diff
@@ -57,17 +60,32 @@ class FileController extends ChangeNotifier {
 
   //------------------------------------------------------ REPEATING TRANSACTIONS ------------------------------------------------------//
 
+  Future<void> readRepTransaction() async {
 
-  updateRepTransaction(int transactionIndex, String transactionName, String transactionDate, List<int> tag, double transactionAmount) async {
-    await FileManager().updateRepTransactionManager(transactionIndex, transactionName, transactionDate, tag, transactionAmount);
-    await refreshTagsAndTransactions();
+    final jsonDataRepTransaction = await FileManager().readFileRepTransactionManager();
+    // If transactions exist and a transaction has tags, add the names of the tags to the Transaction class
+
+    if (jsonDataRepTransaction != null) {
+      listRepTransaction = (jsonDataRepTransaction as List).map((item) => Transaction.fromJson(item as Map<String, dynamic>)).toList();
+      for (var transaction in listRepTransaction) {
+        transaction.transactionTagName.clear();
+        transaction.transactionTagName = transaction.transactionTag.map((index) => listTag[index].tagName).toList();
+      }
+      notifyListeners();
+    }
   }
 
+  updateRepTransaction(int transactionIndex, String transactionName, String transactionDate, List<int> tag, double transactionAmount) async {
+        await FileManager().updateRepTransactionManager(transactionIndex, transactionName, transactionDate, tag, transactionAmount);
+        await refreshTagsAndTransactions();
+      }
+
   deleteRepTransaction(int transactionIndex) async {
-    await FileManager().deleteRepTransactionManager(transactionIndex);
-    await refreshTagsAndTransactions();
```

Comment:
warum der extra indent?

Reply:
wenn du den refresh meinst, der ist in allen drin der sorgt dafür, das die Listen neu geladen werden, ohne das die ganze seite neu geladen werden muss zumindest war das glaube ich Max intention

Reply:
ich meine die einrückung des ciodes. Ich weiß nicht ob du nen Formatter benutzt für den Code, aber ich finde das a) unleserlich und b) komisch von der einrückung her